keyboards/preonic/keymaps/egstad/keymap.c

```diff
@@ -0,0 +1,316 @@
+#include QMK_KEYBOARD_H
+
+
+
+
+
+
+/* LAYER NAMES
+   ==========================================================================
+   1. To create a layer, define a (readable) name and set layer value
+   2. Under "Layer definitions" section, define keymaps
+   ========================================================================== */
+
+#define L_QWERTY  0 
```

Comment:
Optional, but you can use an `enum` here to manage your layers in place of Lines 14-17, which is less labor-intensive.

```c
enum layer_names {
    L_QWERTY,
    L_LOWER,
    L_RAISE,
    L_ADJUST,
};
```

This will be identical to the code you have you when it's compiled.